### PR TITLE
LPS-145792 Export react packages as browser modules

### PR DESCRIPTION
**Pull request by izaera**
This leverages import maps support to export `react` (and others) from `frontend-js-react-web` as ESM modules.

The technique to do that is simple: publish ESM modules that bridge ESM world to AMD loader world, so that no matter if anyone imports `react` from a standard JS Toolkit project, a React fragment, or an ESM module, they all see the very same entity inside the JavaScript engine.

In the future, we may want to provide other frameworks (eg: Angular) in the same way, so that Remote Apps implemented by ESM modules can deduplicate them too.

### CODE REVIEW

**Comment by izaera** [comment]
ci:test:sf

**Comment by izaera** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
I'm terrible at Java, so I just skimmed this, but it looks awesome. The functionality it brings is pretty cool. 

Could I bother you to explain like I'm 5 years old as to how this whole thing functions? I haven't had the need to learn about the differences between module types, let alone how to do what you did here.

**Review by kresimir-coko — APPROVED** [review]
LNHTM 👍

File: modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/servlet/taglib/JSImportmapsExtenderTopHeadDynamicInclude.java [hunk 1]
```diff
@@ -56,28 +60,31 @@ public void include(
...
 		PrintWriter printWriter = httpServletResponse.getWriter();
 
-		printWriter.println("<script type=\"importmap\">");
-		printWriter.println(_importmaps.get());
-		printWriter.println("</script>");
-		printWriter.print("<script src=\"");
+		if (_jsImportmapsConfiguration.enableImportmaps() &&
+			(!_globalImportmaps.isEmpty() || !_scopedImportmaps.isEmpty())) {
+
+			printWriter.println("<script type=\"importmap\">");
```
Comment (julien) [inline modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/servlet/taglib/JSImportmapsExtenderTopHeadDynamicInclude.java:68]: :partying_face:

Comment (izaera) [reply modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/servlet/taglib/JSImportmapsExtenderTopHeadDynamicInclude.java:68]: :sweat_smile:

**Comment by izaera** [comment]
> Could I bother you to explain like I'm 5 years old as to how this whole thing functions? I haven't had the need to learn about the differences between module types, let alone how to do what you did here.

@kresimir-coko I would begin with this -> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules

It explains what ESM modules are and how to use them.

Basically it lets you use `import`, and `export` inside .js files which are directly consumed by the browser, without any need to bundle them with `webpack`. The main goal of this is to be able to build custom elements remote apps that consume react directly from portal (like we did with the bundler, but without any need to use the bundler at all).

**Comment by izaera** [comment]
ci:forward

**Pull request closed without merging**